File: Leetcode/Leetcode-DSA-course/Hashing/383-ransom-note/ransom-note.ts
function canConstruct(ransomNote: string, magazine: string): boolean {
    let freq1 = new Map()
    for (let ch of ransomNote) {
        freq1.set(ch, (freq1.get(ch) || 0) + 1)
    }
    let freq2 = new Map()
    for (let ch of magazine) {
        freq2.set(ch, (freq2.get(ch) || 0) + 1)
    }
    for (let [ch, count] of freq1) {
        if (!freq2.has(ch) || freq2.get(ch) < count) {
            return false;
        }
    }
    return true
};

File: Leetcode/Leetcode-DSA-course/Hashing/383-ransom-note/ransom-note.java
class Solution {
    public boolean canConstruct(String ransomNote, String magazine) {
        Map<Character, Integer> freq1 = new HashMap<>();
        for (char ch : ransomNote.toCharArray()) {
            freq1.put(ch, freq1.getOrDefault(ch, 0) + 1);
        }
        Map<Character, Integer> freq2 = new HashMap<>();
        for (char ch : magazine.toCharArray()) {
            freq2.put(ch, freq2.getOrDefault(ch, 0) + 1);
        }
        for (char ch : freq1.keySet()) {
            if (freq1.get(ch) > freq2.getOrDefault(ch, 0))
                return false;
        }
        return true;
    }
}